dm/syncer/syncer.go

```diff
@@ -1042,10 +1063,79 @@ func (s *Syncer) resetShardingGroup(table *filter.Table) {
 	}
 }
 
+type flushCpTask struct {
+	snapshot SnapshotID
+	wg       *sync.WaitGroup
+	// job version, causality gc max version
+	version int64
+	// extra sharding ddl sqls
+	exceptTables  []*filter.Table
+	shardMetaSQLs []string
+	shardMetaArgs [][]interface{}
+	// result chan
+	resChan chan error
+}
+
+type checkpointFlushWorker struct {
+	input        chan *flushCpTask
+	cp           CheckPoint
+	execError    *atomic.Error
+	afterFlushFn func(task *flushCpTask) error
+}
+
+// Add add a new flush checkpoint job.
+func (w *checkpointFlushWorker) Add(msg *flushCpTask) {
+	w.input <- msg
+}
+
+// Run read flush tasks from input and execute one by one
+func (w *checkpointFlushWorker) Run(ctx *tcontext.Context) {
+	for msg := range w.input {
```

Comment:
```suggestion
	for task := range w.input {
```
maybe use task is better for read here